File: app/src/main/java/ro/scoalainformala/trips/Trip/TripViewModel.java
package ro.scoalainformala.trips.Trip;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class TripViewModel extends AndroidViewModel {

    private TripRepository tripRepository;
    private LiveData<List<Trip>> trips;

    public TripViewModel(@NonNull Application application){
        super(application);

        TripDao tripDao = TripDatabase.getInstance(application)
                .getTripDao();

        tripRepository = new TripRepository(tripDao);

        trips = tripRepository.getTrips();
    }

    public LiveData<List<Trip>> getTrips() {
        return trips;
    }

    public LiveData<List<Trip>> getFavTrips(){
        return tripRepository.getAllFavouriteTrips();
    }

    public void insert(Trip trip) {
        tripRepository.insert(trip);
    }

    public void update(Trip trip){
        tripRepository.update(trip);
    }
}

File: app/src/main/java/ro/scoalainformala/trips/Trip/TripViewModel.kt
package ro.scoalainformala.trips.Trip

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class TripViewModel(application: Application) : AndroidViewModel(application) {
    private val tripRepository: TripRepository
    @JvmField
    val trips: LiveData<List<Trip?>?>?
    init {
        val tripDao : TripDao = TripDatabase.getInstance(application)!!.tripDao
        tripRepository = TripRepository(tripDao)
        trips = tripRepository.trips
    }

    val favTrips: LiveData<List<Trip?>?>?
        get() = tripRepository.allFavouriteTrips

    fun insert(trip: Trip?) {
        tripRepository.insert(trip)
    }

    fun update(trip: Trip?) {
        tripRepository.update(trip)
    }
}
